zenpy/lib/api.py

```diff
@@ -1027,28 +1028,33 @@ def upload(self, fp, token=None, target_name=None, content_type=None):
                                         target_name=target_name,
                                         content_type=content_type)
 
-    def download(self, attachment_id, destination):
+    def download(self, attachment_id, destination=None):
```

Comment:
Might be better to allow passing a BytesIO instance here instead of checking for None and creating one.

Reply:
It sure might be, though the current logic supports BytesIO or location on disk, thus currently spares the user for one additional passing argument.

Do you suggest me updating this PR with a more generic solution?

Reply:
Well explicit better than implicit etc etc :)

That said if you like this version better just resolve this conversation and I'll merge as is a nice change either way.